### PR TITLE
Add a chapter on useful `@rustbot` commands

### LINKED ISSUE #855: Add section on `@rustbot modify labels`
This is [really hard to find for new contributors](https://rust-lang.zulipchat.com/#narrow/stream/242269-t-release.2Ftriage/topic/Label.20switching.20for.20outsiders.20is.20hard): https://github.com/rust-lang/triagebot/wiki/Labeling

### PR DESCRIPTION
**Pull request by LeSeulArtichaut**
Closes #855 (cc @jyn514)

### CODE REVIEW

File: src/contributing.md [hunk 1]
```diff
@@ -154,12 +154,17 @@ rebase when bringing the latest changes from the master branch to your feature
 branch.  Also, please make sure that fixup commits are squashed into other
 related commits with meaningful commit messages.
 
+If you encounter merge commits, your PR will get marked as `S-waiting-on-author`.
```
Comment (camelid) [inline src/contributing.md:157]:
```suggestion
If you encounter merge conflicts, your PR will get marked as `S-waiting-on-author`.
```

File: src/contributing.md [hunk 2]
```diff
@@ -413,8 +418,8 @@ still valid. Load up an older issue, double check that it's still true, and
 leave a comment letting us know if it is or is not. The [least recently
 updated sort][lru] is good for finding issues like this.
 
-Contributors with sufficient permissions on the Rust repo can help by adding
-labels to triage issues:
+[Thanks to `@rustbot`][rustbot], anyone can help triaging issues by adding
```
Comment (camelid) [inline src/contributing.md:421]:
```suggestion
[Thanks to `@rustbot`][rustbot], anyone can help triage issues by adding
```

File: src/rustbot.md [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+
+`@rustbot` exposes a command that allows anyone to assign an issue to themselves.
+If you see an issue you want to work on, you can send the following message as a
+comment on the issue at hand:
+
+    @rustbot claim
+
+If you are a member of the `rust-lang` organization, `@rustbot` will assign the
+issue to you. Otherwise, `@rustbot` will assign itself as a placeholder and edit
+the top comment to reflect the fact that the issue is now assigned to you.
```
Comment (camelid) [inline src/rustbot.md:18]: I don't think this is true anymore; I think rustbot will assign you regardless of whether you're part of the org. For example, I'm not a member of rust-lang, but rustbot assigns me directly. Cc @Mark-Simulacrum - am I understanding it correctly?

Comment (jyn514) [reply src/rustbot.md:18]: I've definitely seen it assign itself before, not the person claiming. Not sure why it assigns you but not others though.

Comment (camelid) [reply src/rustbot.md:18]: Maybe it likes me ;)

I haven't seen it do that in a while though.

File: src/rustbot.md [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+## Issue relabeling
+
+Changing labels for an issue or PR is also normally reserved to members of the
+organization. However, `@rustdoc` allows you to relabel an issue yourself, only
+with a few restrictions. This is mostly useful in two cases:
+
+**Helping with issue triage**: Rust's issue tracker has more than 5,000 open
+issues at the time of this writing, so labels are the most powerful tool that we
+have to keep it as tidy as possible. You may not spend hours in the issue tracker
+to triage issues, but if you open an issue, you should feel free to label it if
```
Comment (jyn514) [inline src/rustbot.md:33]: 'may not' means "you can't", but I think you mean "you shouldn't".
```suggestion
have to keep it as tidy as possible. You don't need to spend hours in the issue tracker
to triage issues, but if you open an issue, you should feel free to label it if
```

**Review by jyn514 — APPROVED** [review]
LGTM with nits fixed, thanks for adding this!

File: src/rustbot.md (hunk 3 above)
Comment (camelid) [reply src/rustbot.md:18]: I just looked and I've seen `@rustbot claim` assign issues to `@lcnr`, who is not a GitHub org member, and `@jumbatm`, who is likewise not an org member. It definitely used to assign itself, but the most recent issue `rustbot` is assigned to was opened July 31.

Comment (jyn514) [reply src/rustbot.md:18]: (how is lcnr not an org member? they've implemented like half of min_const_generics 😆 )

Comment (LeSeulArtichaut) [reply src/rustbot.md:18]: They are now :P
Btw they were member or the Prioritization WG which is "enough" to be part or the GitHub org

Comment (Mark-Simulacrum) [reply src/rustbot.md:18]: IIRC, triagebot will assign you directly if you either are an org member or commented on the issue/PR where the assignment happens. But I would consider this an implementation detail - does it not suffice to say that sometimes you will be assigned directly and sometimes indirectly?

Comment (LeSeulArtichaut) [reply src/rustbot.md:18]: Thanks for the clarification! I think you’re right, just saying that you might be assigned indirectly is the best way of presenting things indeed

File: src/contributing.md [hunk 5]
```diff
@@ -435,7 +440,9 @@ labels to triage issues:
 
 * Red, **I**-prefixed labels indicate the **importance** of the issue. The
   [I-nominated][inom] label indicates that an issue has been nominated for
-  prioritizing at the next triage meeting.
+  prioritizing at the next triage meeting. Similarely, the [I-prioritize][ipri]
```
Comment (tshepang) [inline src/contributing.md:443]:
```suggestion
  prioritizing at the next triage meeting. Similarly, the [I-prioritize][ipri]
```

File: src/rustbot.md [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+the top comment to reflect the fact that the issue is now assigned to you.
+
+If you want to unassign from an issue, `@rustbot` has a different command:
+
+    @rustbot release-assignment
+
+## Issue relabeling
+
+Changing labels for an issue or PR is also normally reserved to members of the
+organization. However, `@rustdoc` allows you to relabel an issue yourself, only
```
Comment (tshepang) [inline src/rustbot.md:27]:
```suggestion
organization. However, `@rustbot` allows you to relabel an issue yourself, only
```

File: src/rustbot.md [hunk 7]
```diff
@@ -0,0 +1,62 @@
...
+**Helping with issue triage**: Rust's issue tracker has more than 5,000 open
+issues at the time of this writing, so labels are the most powerful tool that we
+have to keep it as tidy as possible. You may not spend hours in the issue tracker
+to triage issues, but if you open an issue, you should feel free to label it if
+you are comfortable with doing it yourself.
+
+**Updating the status of a PR**: we use "status labels" to reflect the status of
+PRs. For example, if your PR has merge conflicts, it will automatically be assigned
+the `S-waiting-on-author`, and reviewers might not review it until you rebase your
+PR. Once you did rebase your branch, you should change the labels yourself to remove
```
Comment (tshepang) [inline src/rustbot.md:39]:
```suggestion
PR. Once you do rebase your branch, you should change the labels yourself to remove
```

File: src/rustbot.md [hunk 8]
```diff
@@ -36,7 +36,7 @@ you are comfortable with doing it yourself.
 **Updating the status of a PR**: we use "status labels" to reflect the status of
```
Comment (camelid) [inline src/rustbot.md:36]: It's fine either way, but do you want to do this?

```suggestion
**Updating the status of a PR**: We use "status labels" to reflect the status of
```

**Comment by LeSeulArtichaut** [comment]
Resolved all the reviews, this should be ready for merging now :)

File: src/rustbot.md [hunk 9]
```diff
@@ -13,9 +13,10 @@ comment on the issue at hand:
 
     @rustbot claim
 
-If you are a member of the `rust-lang` organization, `@rustbot` will assign the
-issue to you. Otherwise, `@rustbot` will assign itself as a placeholder and edit
-the top comment to reflect the fact that the issue is now assigned to you.
+This will tell `@rustbot` to assign the issue to you if it has no assignee yet.
+Note that because of some GitHub restrictions, you may be assigned indireclty,
```
Comment (jyn514) [inline src/rustbot.md:17]:
```suggestion
Note that because of some GitHub restrictions, you may be assigned indirectly,
```

**Comment by LeSeulArtichaut** [comment]
Wait. I made a typo in the message of a commit fixing a typo...

**Comment by jyn514** [comment]
> Wait. I made a typo in the message of a commit fixing a typo...

We have to go deeper ... typos in coments about making typos while fixing typos.

**Comment by LeSeulArtichaut** [comment]
Fixed x)

**Comment by jyn514** [comment]
Looks good to me, we can always make improvements in later PRs :)

**Pull request merged**